frontend/src/worker/worker.ts

```diff
@@ -0,0 +1,36 @@
+import { AnnouncementDetail, UserAnnouncement, Umbra } from '@umbra/umbra-js';
+import { getAddress } from 'src/utils/ethers';
+
+export const filterUserAnnouncements = (
+  spendingPublicKey: string,
+  viewingPrivateKey: string,
+  announcements: AnnouncementDetail[],
+  progress: (percentage: number) => void,
+  completion: (userAnnouncements: UserAnnouncement[]) => void
+) => {
+  const userAnnouncements: UserAnnouncement[] = [];
+  let index = 0;
+  const chunk = 10;
```

Comment:
I chose 10 because that was the point at which it started feeling smoothest. If we wanted to really tune this, we would test it across different devices, but I don't think it's a big enough deal.

I don't think we need to make this an ENV var because it's not something that will need to be changed based on changing external conditions. I.E. the experience is the same regardless of the number of events we need to scan.